awserr.go

```diff
@@ -42,3 +36,48 @@ func IsAWSErrExtended(err error, code string, message string, origErrMessage str
 
 	return false
 }
+
+// IsAWSErrCode returns true if the error matches all these conditions:
+//  * err is of type awserr.Error
+//  * Error.Code() equals code
+func IsAWSErrCode(err error, code string) bool {
+	var awsErr awserr.Error
+	if errors.As(err, &awsErr) {
+		return awsErr.Code() == code
+	}
+	return false
+}
+
+// IsAWSErrCodeContains returns true if the error matches all these conditions:
+//  * err is of type awserr.Error
+//  * Error.Code() contains code
+func IsAWSErrCodeContains(err error, code string) bool {
+	var awsErr awserr.Error
+	if errors.As(err, &awsErr) {
+		return strings.Contains(awsErr.Code(), code)
+	}
+	return false
+}
+
+// IsAWSErrCodeMessageContains returns true if the error matches all these conditions:
+//  * err is of type awserr.Error
+//  * Error.Code() equals code
+//  * Error.Message() contains message
+func IsAWSErrCodeMessageContains(err error, code string, message string) bool {
+	var awsErr awserr.Error
+	if errors.As(err, &awsErr) {
+		return awsErr.Code() == code && strings.Contains(awsErr.Message(), message)
+	}
+	return false
+}
+
+// IsAWSErrRequestFailureStatusCode returns true if the error matches all these conditions:
```

Comment:
This comment discouraging use should also be brought over:

```go
// It is always preferable to use isAWSErr() except in older APIs (e.g. S3)
// that sometimes only respond with status codes.
```

Reply:
Done.